File: Python/백준/Bronze/2798. 블랙잭/블랙잭.py
N, M = map(int,input().split())
A = sorted(list(map(int,input().split())))
MAX = 0
for i in range (N):
    for j in range(i+1,N):
        for k in range (j+1,N):
            SUM = A[i]+A[j]+A[k]
            if  MAX < SUM <= M:
                MAX = SUM
print(MAX)

File: Java/백준/Bronze/2798. 블랙잭/블랙잭.java
import java.io.IOException;
import java.util.*;

class Main {
    public static void main(String[] args) throws IOException {
        Scanner in = new Scanner(System.in);
        int N = in.nextInt();
        int M = in.nextInt();
        int[] Card = new int[N];
        for (int i=0; i<N; i++){
            Card[i] = in.nextInt();
        }
        int max = 0;
        for (int i=0; i<N-2; i++){
            for (int j=i+1; j<N-1; j++){
                for (int k=j+1; k<N; k++){
                    int sum = Card[i]+Card[j]+Card[k];
                    if (max<sum && sum<=M){
                        max = sum;
                    }
                }
            }
        }
        System.out.print(max);
    }
}